Orange Pi 4 refactor to mainline

Shaggy013 opened the pull request:
Orange Pi4 updates to make the dts more common to mainline and added some features

like spi boot
hdmi outpit in uboot
type c fix  (@tonymac this is new dp fix )

refactor kernel dts dot make it same as the rest of the boards.

# Description

Changes the dts of uboot and kernel to be more mainline 
can be build with blobs or atf bot work 

added spi boot
Usb3 is seen with uboot  
Hdmi output on uboot
old uboot has lot of wrong pin , a mipi switch is not a usbc switch etc 
only problem is how to switch fusb on but that have all boards


and kernel side it 

fixed type-c (@tonymac this is the replacement of extcon)
and refactor to mainline and alphabet 



# How Has This Been Tested?

Tested on Orange Pi 4 
on sd and spi 

with uboot debug 
and comapring dmesg of kernel 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

Tonymac32 commented:
Can anyone test this to verify the display port functionality, I don't have this board.  It is a required workaround to my knowledge that an extcon is still required for DP alt mode to function, at least as of 5.10.  If that is handled here in some way I'm not noticing, then great, I'll roll it out to the other boards.

Shaggy013 commented:
it is the same as other patches around for that fuzzyusb, for the kernel it works for me to see a usb flash have no dp connector yet 

for example 
https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/rockchip/rk3399-pinebook-pro.dts
https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/rockchip/rk3399-orangepi.dts
 
so it does switch to other modes maybey your lucky it fix also dp for you ,for me it was to see usb drives flash and sata

only only not get that fuzzy usb to powerup at uboot for the kernel it did fix fo me

piter75 commented on config/boards/orangepi4.conf:
```diff
@@ -1,8 +1,9 @@
 # Rockchip RK3399 hexa core 4GB RAM SoC GBE eMMC USB3 USB-C WiFi/BT
 BOARD_NAME="OrangePi 4"
-BOARDFAMILY="rk3399"
+BOARDFAMILY="rockchip64"
```
Why moving it to `rockchip64` family?
Does it have anything for Orange Pi 4 that `rk3399` lacks?
This move definitely borks `legacy` branch for Orange Pi 4

Shaggy013 replied:
ah is thats why he was in rk3399 

no it is just a rk3399 just like the rockchip64

did do dev no legacy but it not matter where it is it 

thnx

Tonymac32 commented:
For the other USB patches yes it matches the mainline, however that did not
work with the display port in my tests, and I had to employ the workaround
found in our device trees for the helios64 and the pinebook pro, which is
also found in Manjaro as I was researching.  You also have cdn_DP disabled
in the device tree, enable it and check the dmesg, I believe you will get
an error that no ports have been assigned.

On Tue, Feb 23, 2021, 1:23 PM Shaggy013 <notifications@github.com> wrote:

> *@Shaggy013* commented on this pull request.
> ------------------------------
>
> In config/boards/orangepi4.conf
> <https://github.com/armbian/build/pull/2658#discussion_r581283641>:
>
> > @@ -1,8 +1,9 @@
>  # Rockchip RK3399 hexa core 4GB RAM SoC GBE eMMC USB3 USB-C WiFi/BT
>  BOARD_NAME="OrangePi 4"
> -BOARDFAMILY="rk3399"
> +BOARDFAMILY="rockchip64"
>
> ah is thats why he was in rk3399
>
> no it is just a rk3399 just like the rockchip64
>
> did do dev no legacy but it not matter where it is it
>
> thnx
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/armbian/build/pull/2658#discussion_r581283641>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFATFAXGP2CJ4ISDH44EALDTAPXD5ANCNFSM4YCV3XYA>
> .
>

Shaggy013 commented:
well the dmesg is on the forum so you can check yourself 

i disabled &cdn_dp in favor of hdmi uboot and kernel 

you can enable &cdn_dp but then you have to disable hdmi in uboot it is just what you want proberly there are better fixes 

and it is my first contribution to the git so please kill me :)

it fixes a lot for the orange pi 4 and 4bf or example  spi and usb3 in uboot  

and kernel is just put on alphabetic and the patch that solves typec for orange pi 

and just post what is possible , and did dp or typec work before on the orange pi 4 on mainline kernel ?

Shaggy013 commented:
fixed led in uboot
added led powerkey and recovery key to uboot

only have to solve why CONFIG_OHCI_HCD breaks uboot build
and why the type c port get no power ( fusb ? , regulator?,gpio ? )

and how to get the updates here :)

Shaggy013 commented:
Yeah found the regulator problem .
had even to put my scope to it to solve it 
created a board for the opi 4 and created the switch there just like pinebook helios etc only the one for 1v8 

and now it turns on on boot 

only have to figure out how to put on git

AndreiCherniaev commented:
> Orange Pi4 updates to make the dts more common to mainline and added some features
> 
> like spi boot
> hdmi outpit in uboot
> type c fix (@TonyMac this is new dp fix )
> 
> refactor kernel dts dot make it same as the rest of the boards.
> # Description
> 
> Changes the dts of uboot and kernel to be more mainline
> can be build with blobs or atf bot work
> 
> added spi boot
> Usb3 is seen with uboot
> Hdmi output on uboot
> old uboot has lot of wrong pin , a mipi switch is not a usbc switch etc
> only problem is how to switch fusb on but that have all boards
> 
> and kernel side it
> 
> fixed type-c (@TonyMac this is the replacement of extcon)
> and refactor to mainline and alphabet
> # How Has This Been Tested?
> 
> Tested on Orange Pi 4
> on sd and spi
> 
> with uboot debug
> and comapring dmesg of kernel
> # Checklist:
> 
>     * [ ]  My code follows the style guidelines of this project
> 
>     * [ x] I have performed a self-review of my own code
> 
>     * [ ]  I have commented my code, particularly in hard-to-understand areas
> 
>     * [ x] I have made corresponding changes to the documentation
> 
>     * [ x] My changes generate no new warnings
> 
>     * [ ]  Any dependent changes have been merged and published in downstream modules

Are you those bad boy that take my [uart4](https://github.com/orangepi-xunlong/orangepi-build/issues/4#issuecomment-823934575) because of spi1?

Shaggy013 commented:
Nope uart4 is disabled by default

Only added i2c2

And spi1 was already there

And corrected the port numbers
